watertap/costing/multiple_choice_costing_block.py

```diff
@@ -0,0 +1,209 @@
+        add_object_reference(self, "costing_package", fcb)
+        add_object_reference(self, "unit_model", unit_model)
+
+        self.costing_blocks = pyo.Block(self.config.costing_blocks)
+        self.costing_selector = pyo.Param(
+            self.config.costing_blocks, domain=pyo.Boolean, default=0, mutable=True
+        )
+
+        if self.config.initial_active_block is None:
+            for k in self.config.costing_blocks:
+                self.costing_selector[k].set_value(1)
+                break
+        else:
+            self.costing_selector[self.config.initial_active_block].set_value(1)
+
+        # Get costing method if not provided
+        for k, val in self.config.costing_blocks.items():
+
+            # if we get a callable, just use it
+            if callable(val):
+                method = val
+                kwds = {}
+            # else we'll assume it's a dictionary with keys
+            # `costing_method` and potentially `costing_method_arguments`.
+            # We raise an error if we find something else.
+            else:
+                for l in val:
+                    if l not in ("costing_method", "costing_method_arguments"):
+                        raise RuntimeError(
+                            f"Unrecognized key {l} for costing block " f"{k}."
+                        )
+                try:
+                    method = val.get("costing_method")
+                except KeyError:
+                    raise KeyError(
+                        f"Must specify a `costing_method` key for costing "
+                        f"block {k}."
+                    )
+                kwds = val.get("costing_methods_arguments", {})
+
+            blk = self.costing_blocks[k]
+
+            # Assign object references for costing package and unit model
+            add_object_reference(blk, "costing_package", fcb)
+            add_object_reference(blk, "unit_model", unit_model)
+
+            # Call unit costing method
+            method(blk, **kwds)
+
+            # Check that costs are Vars and have lower bound of 0
+            cost_vars = DefaultCostingComponents
+            for v in cost_vars:
+                try:
+                    cvar = getattr(self, v)
+                    if not isinstance(cvar, pyo.Var):
+                        raise TypeError(
+                            f"{unit_model.name} {v} component must be a Var. "
+                            "Please check the costing package you are using to "
+                            "ensure that all costing components are declared as "
+                            "variables."
+                        )
+                    elif cvar.lb is None or cvar.lb < 0:
+                        _log.warning(
+                            f"{unit_model.name} {v} component has a lower bound "
+                            "less than zero. Be aware that this may result in "
+                            "negative costs during optimization."
```

Comment:
We could have negative components to account for revenue. Is that included in `DefaultCostingComponents` ?

Reply:
Yes -- this line is actually from IDAES such that we replicate the existing behavior, warts and all. I don't think the user should notice a difference between using a `UnitModelCostingBlock` or a (tentatively named) `MultipleChoiceCostingBlock` with a single method and get different behavior.